### ISSUE TITLE
Get "MySQLDictionarySource: Cannot find table in SHOW TABLE STATUS result" error when restart server

### ISSUE THREAD

**Issue by v-electrolux** [ClickHouse/ClickHouse#45885]
When restarting server, get error in logs:
```
2023.02.01 09:59:16.254642 [ 493 ] {} <Error> MySQLDictionarySource: Cannot find table in SHOW TABLE STATUS result
```
But no errors when dictionary update its content.

Dictionary definition
```
CREATE DICTIONARY test.test_dict
(
    `account_name` String,
    `account_uuid` Int32 INJECTIVE
)
PRIMARY KEY account_name
SOURCE(MYSQL(
  PORT 3306
  USER 'test_user'
  PASSWORD 'test_pass'
  REPLICA (HOST 'mysql.shard-01.replica' PRIORITY 1)
  DB 'test'
  QUERY 'SELECT t1.account_name, t2.account_uuid FROM table1 t1 LEFT JOIN table2 t2 ON t1.id = t2.account_id'
  INVALIDATE_QUERY 'SELECT max(UPDATE_TIME) FROM information_schema.tables WHERE TABLE_SCHEMA = "test" AND (TABLE_NAME = "table1" OR TABLE_NAME = "table2") GROUP BY TABLE_SCHEMA'
  FAIL_ON_CONNECTION_LOSS 'true'))
LIFETIME(MIN 60 MAX 120)
LAYOUT(COMPLEX_KEY_HASHED());
```
ClickHouse version 22.5.1.2079 (official build)

As we can see, dictionary has INVALIDATE_QUERY and QUERY options, and does not have TABLE option.
So I think problem in function
https://github.com/ClickHouse/ClickHouse/blob/26647d3ccab72a7a6f84fd57b9e571b8d24c3d8c/src/Dictionaries/MySQLDictionarySource.cpp#L198
or/and
https://github.com/ClickHouse/ClickHouse/blob/26647d3ccab72a7a6f84fd57b9e571b8d24c3d8c/src/Dictionaries/MySQLDictionarySource.cpp#L207
which called when restart server. Because when dictionary checks, should it update content or not, it calls function
https://github.com/ClickHouse/ClickHouse/blob/26647d3ccab72a7a6f84fd57b9e571b8d24c3d8c/src/Dictionaries/MySQLDictionarySource.cpp#L231
which checks INVALIDATE_OPTION exist or not

**Comment by tavplubix** [comment]
First of all, version 22.5 is obsolete, and you should upgrade. 

This error can make dictionary reloading less optimal, but it does not look like a major issue, everything should still work correctly. 

The problem is that `configuration.table` is empty because a table name is not specified in the dictionary definition and `QUERY` clause is specified instead. It's not trivial to get data modification time in this case, and it was not implemented in https://github.com/ClickHouse/ClickHouse/pull/26995

https://github.com/ClickHouse/ClickHouse/blob/36e34d8cc610e3afc25b5f5bc79e3775ee93be75/src/Dictionaries/MySQLDictionarySource.cpp#L297-L327

**Comment by v-electrolux** [comment]
> This error can make dictionary reloading less optimal, but it does not look like a major issue, everything should still work correctly.

That's right, it is not a major issue, everything works fine. Just pollutes logs a little.

> The problem is that configuration.table is empty because a table name is not specified in the dictionary definition and QUERY clause is specified instead. It's not trivial to get data modification time in this case, and it was not implemented in https://github.com/ClickHouse/ClickHouse/pull/26995

Correct. But user, when set INVALIDATE_QUERY expects that such query will be used every time when ClickHouse need to get data modification time. It seems, that function
https://github.com/ClickHouse/ClickHouse/blob/26647d3ccab72a7a6f84fd57b9e571b8d24c3d8c/src/Dictionaries/MySQLDictionarySource.cpp#L288
must never called, when INVALIDATE_QUERY is set. Because if developers find a solution when we can set both QUERY and TABLE, we get two different methods to get data modification time executed when ClickHouse starts and when dictionary invalidate itself by timer

**Comment by sergio-ropero** [comment]
Any news on this issue?

Still happening in latest versions (we have installed version 24.8.5.115).

The problem is that we have several dictionaries with a short refresh rate so we are not only polluting logs, but sending unneeded data through network to monitor the system (unwanted increase in billing).

Also this impacts the log file rotation, losing all the information from the logs in a short period of time.

**Comment by den-crane** [comment]
* https://github.com/ClickHouse/ClickHouse/pull/73914

it will be released with 25.1
